conbench/entities/benchmark_result.py

```diff
@@ -35,6 +35,9 @@ class BenchmarkResult(Base, EntityMixin):
     data = Nullable(postgresql.ARRAY(s.Numeric), default=[])
     times = Nullable(postgresql.ARRAY(s.Numeric), default=[])
     case = relationship("Case", lazy="joined")
+    # optional info at the benchmark level (i.e. information that isn't a tag that should create a separate case, but information that's good to hold around like links to logs)
```

Comment:
This name is not great, but I don't want to write the giant migration that would be required to do the following:

I think we should have at the benchmark level (i.e. that will vary with each benchmark): 
* `case` (which generally come from tags...)
* `optional_info` (or we could call it even more explicitly: `additional_info` or `additional_benchmark_info`)

And then across benchmarks within a run have :
* `context` (just like we have now)
* `context_optional_info` (or `context_additional_info` or whatever to match ^^^) that is our `info` now

This way it's clear(er) that each benchmark has some info about it that matters for performance but is unique to the benchmark (the `case` info), some info that doesn't matter for performance but is unique to the benchmark (the `optional_info`), and then there is info that matters for performance but is shared across benchmarks (`context`) and info that doesn't matter for performance but is shared across benchmarks (`context_optional_info`)

Reply:
Trying to understand what's supposed to go where here. 

Firstly, isn't `context` constant across many runs? Like `benchmark_language` and `compiler_flags` shouldn't change often. So `context_additional_info` doesn't make much sense to me exactly when we're talking about something that can change at the run level. Or am I wrong and is `context_additional_info` supposed to be as invariant as `context`? ...because AFAIK we aren't necessarily using it that way right now.

My understanding:

- `optional_info` is non-history-deciding metadata that can vary per run of a `case` for a given benchmark
- `info` (or whatever; called "Additional Info" in the website) should stay the same across all cases for all benchmarks in a run? —but that's weird, as you can also pass `info` when creating a run, and this (AFAIK) gets stored on `BenchmarkResult`

It seems like the missing level of would then be for info that's constant for all cases of a benchmark, but varies between benchmarks in a run. Maybe `Run.info` should be the one we show under `context` (which should really probably live on the run, too, in theory), and `BenchmarkResult.info` should be constant for all cases of a benchmark, and `optional_info` can vary by case? ...though I'm not sure that's practical given how we don't usually manually create `Run`s (for now).

All that said, I'm not sure how much of metadata we would have to populate each of these. Or are there actually any restrictions on `BenchmarkResult.info` that we couldn't overload the existing field to optionally provide multiple levels of info, nested (e.g. `"info": {"case": {...}, "benchmark": {...}, "run": {...}}`.

As a user, we need to clearly document what kind of metadata should get stuck where, because it's far from obvious from the schema.

Reply:
I guess there are potentially four levels of info? `context_info`, `run_info`, `benchmark_info`, and `case_info`. Except where they live in the database would be a little weird given we're passing most of these in with a case result.